pkg/controller/operators/catalog/operator.go

```diff
@@ -615,10 +615,16 @@ func (o *Operator) syncRegistryServer(logger *logrus.Entry, in *v1alpha1.Catalog
 	// Registry pod hasn't been created or hasn't been updated since the last configmap update, recreate it
 	logger.Debug("ensuring registry server")
 
-	if err := srcReconciler.EnsureRegistryServer(out); err != nil {
-		syncError = fmt.Errorf("couldn't ensure registry server - %v", err)
-		out.SetError(v1alpha1.CatalogSourceRegistryServerError, syncError)
-		return
+	err = srcReconciler.EnsureRegistryServer(out)
+	if err != nil {
+		if _, ok := err.(reconciler.UpdateNotReadyErr); ok {
+			logger.Debug("requeueing registry server for catalog update check: update pod not yet ready")
+			o.catsrcQueueSet.RequeueAfter(out.GetNamespace(), out.GetName(), 10*time.Second)
```

Comment:
Why don't we return if the Update is not ready?

Reply:
Wasn't sure if we wanted to the code below ie the `err := o.sources.Remove(sourceKey)` to run, but looking at it now I'm pretty sure its better to return early